// destcloud/utils/logger.js
var express = require('express');
var fs = require('fs');
var winston = require('winston');
winston.emitErrs = true;
var moment = require('moment');

var LOG_DIR = '/var/log/destcloud1/';

var app = express();
if (app.get('env') == 'development') {
    LOG_DIR = './logs/';
}

fs.mkdir(LOG_DIR, function(err) {
    if (err) {
	if (err.code == 'EACCES') {
	    console.error("Cannot access " + LOG_DIR);
	    console.error('logger requires elevated privileges');
	    process.exit(1);
	}
	console.log(LOG_DIR + ' exists.');
    }
});

var LOG_FILE = LOG_DIR + 'destcloud-logs.log';
console.log("destcloud1 log file: " + LOG_FILE);

var logger = new winston.Logger({
    transports: [
	new winston.transports.File({
	    level: 'info',
	    filename: LOG_FILE,
	    handleExceptions: true,
	    humanReadableUnhandledException: true,
	    json: false,
	    timestamp: function() {
		return moment().format('YYYY/MM/DD HH:mm:ss.SSS');
	    },
	    maxsize: 5242880,	// 5MB
	    maxFile: 5,
	    zippedArchive: true,
	    colorize: false
	}),
	new winston.transports.Console({
	    level: 'silly',
	    handleExceptions: true,
	    json: false,
	    showLevel: false,
	    colorize: true
	})
    ],
    exitOnError: false
});

module.exports = logger;
module.exports.stream = {
    write: function(message, encoding) {
	logger.info(message);
    }
};
